src/gui/sdlmain.cpp

```diff
@@ -624,7 +608,15 @@ static SDL_Window * GFX_SetSDLWindowMode(Bit16u width, Bit16u height, bool fulls
 		SDL_SetWindowFullscreen(sdl.window,
 		                        sdl.desktop.full.display_res ? SDL_WINDOW_FULLSCREEN_DESKTOP : SDL_WINDOW_FULLSCREEN);
 	} else {
-		SDL_SetWindowSize(sdl.window, width, height);
+		// If you feel inclined to remove this SDL_SetWindowSize call
+		// (or further modify when it is being invoked), then make sure
+		// window size is updated AND content is correctly clipped when
+		// emulated program changes resolution with various
+		// windowresolution settings (!).
+		if (sdl.desktop.window.use_original_size &&
+		    !sdl.desktop.switching_fullscreen)
+			SDL_SetWindowSize(sdl.window, width, height);
```

Comment:
Might be out of scope, but an area I've wanted to fix is the `sdl.desktop.window.<height/width>` variable; which is  always incorrect or even invalid (such as being zeros). 

In this case, we're setting the window-mode size, `SDL_SetWindowSize()`, but we're not taking in `sdl.desktop.window.<dims>`, nor are we updating it to reflect the as-set Window size.  So that would be one improvement.

Reply:
Only when implementing this PR I finally understood how `sdl.desktop.window.width/height` work… as it turns out, value 0 is not invalid (!) - it's an indication, that user selected `windowresolution=original` - I did not understood that, that's why I added `window_use_original_size` boolean some time ago (which I still think works better than having 0 treated as undocumented special value). The actual logic for turning 0 into the proper window size value is implemented in function `GFX_SetupWindowScaled` (which is totally a private function to sdlmain / GFX "module", so GFX_ name is misleading).

So… if we updated `sdl.desktop.window.width/height`, then clipping code gets confused after first resolution change coming from emulated application (I use Norton Commander for testing it - default screensaver "Floppies" triggers change from 80x25 text mode to 320x200 graphics mode).

Reply:
Yes; fully-agree that zero is more harmful and incorrect than using `original`, which atleast conveys more meaning.  _(Also agree that tackling this properly should be post-0.75; so just discussing ideas now :-)_

But yeah, ultimately, if we've got a Window, then it's got some dimensions, regardless of how they translate into the DOS world (80x25 or a graphical mode).  In SDL-land, it's just a surface with pixels.  So that's what I'm after, where ever we want to store it.

I hit the need for the current Window-mode dimensions when trying to fix the bug you hit when coming out of fullscreen mode and then middle clicking.  Discovering the Window-mode dimensions just after the mode-flip will let me warp the mouse cursor to the same relative position inside the Window to the same relative location the mouse was when in full-screen.

Reply:
Warping mouse cursor seems like a great idea to solve that bug :) see my comment below for a hint about obtaining up-to-date window size.